exatomic/core/basis.py

```diff
@@ -126,12 +129,18 @@ def spherical_by_shell(self, program, spherical=True):
     def functions_by_shell(self):
         """Return a series of n functions per (set, L).
         This does not include degenerate functions."""
-        return self.groupby(['set', 'L'])['shell'].nunique()
+        names = ['set', 'L']
```

Comment:
Maybe worth making this a module/class level attribute since it appears in multiple methods